File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public static int solution(int n) {
		int ans = 0;

		for (int i = 0; i <= n; i++) {
			if (isPrime(i)) ans++;
		}
		return ans;
	}

	private static boolean isPrime(int n) {
		if (n < 2) return false;
		for (int i = 2; i <= Math.sqrt(n); i++) {
			if (n % i == 0) return false;
		}

		return true;
	}
}

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import kotlin.math.sqrt

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val arr = br.readLine().split(" ").map { it.toInt() }.toIntArray()
    var cnt = 0

    for (i in 0 .. arr.lastIndex) {
        if (isPrime(arr[i])) cnt++
    }
    println(cnt)
}
private fun isPrime (num : Int): Boolean {
    if (num == 1) return false

    for (i in 2 .. sqrt(num.toDouble()).toInt()) {
        if (num % i == 0) return false
    }

    return true
}